brain/compartmentReportReader.cpp

```diff
@@ -178,6 +182,45 @@ std::future<CompartmentReportFrames> CompartmentReportView::load(float start,
 
 std::future<CompartmentReportFrames> CompartmentReportView::loadAll()
 {
+    if (_impl->report->getGIDs().size() == 1)
+    {
+        auto report = _impl->report;
+        auto id = *_impl->report->getGIDs().begin();
+
+        double start = _impl->report->getStartTime();
+        double end = _impl->report->getEndTime();
+        double timestep = _impl->report->getTimestep();
+        auto loadFrameTask = [start, end, timestep, report, id] {
+
+            CompartmentReportFrame frame;
+
+            try
+            {
+                auto data = report->loadNeuron(id);
+                if (data)
+                    frame._impl->data = std::move(*data);
```

Comment:
This doesn't look correct to me. I think that the only proper way of using loadNeuron without having to do an extra copy here (and another one in the wrapping) is to get rid of the ComparmentReportFrame class and return the data directly in a matrix-like object with the same layout expected by the numpy array. Here you're passing an array of float point values for all frames of a neuron into a single frame, which conceptually is wrong.
Note that the function returns different results for the single cell case depending on the implementation supporting loadNeuron or not and that shouldn't be the case.
For this PR I'd focus on the Python side and leave this for later because I don't see how to do it efficiently without changes in the C++ API.